src/prefect/backend/flow_run.py

```diff
@@ -583,13 +583,16 @@ def from_flow_run_id(
             A populated `FlowRunView` instance
         """
         flow_run_data = cls._query_for_flow_run(where={"id": {"_eq": flow_run_id}})
+        _cached_task_runs = list(_cached_task_runs or [])
 
         if load_static_tasks:
             task_run_data = TaskRunView._query_for_task_runs(
                 where={
                     "map_index": {"_eq": -1},
                     "flow_run_id": {"_eq": flow_run_id},
+                    "id": {"_nin": [view.task_run_id for view in _cached_task_runs]},
```

Comment:
Forgive my v1 ignorance, but I assume that `_nin` means `not in`? So this is effectively saying "Get all of the task runs associated with this flow run, except for the ones that are cached."

Reply:
Yep! We only cache finished task runs so excluding them from the query results will reduce the amount of traffic with no change in behavior.